### PR TITLE
Adui 5371 investigate move to tsc/rollup instead of webpack PART 1

### PR DESCRIPTION
**Pull request by j-mcgregor**
This PR will fail. This is expected.

### Proposed Changes

Part 1 - importing Jest config, removing Jasmine, Karma, Webpack

### Potential Breaking Changes

Too many to list - Jest uses Jasmine behind the scenes but everything else is different
This PR will point to Master. A separate PR containing the test fixes will be made and point to this branch. There will be hundreds of files in that one running around 3,500 UTs. 

Most of those change will be slight variations (eg `jasmine.spy() -> jest.spyOn()`, but many won't be 

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        'ts-jest': {
+            // relative path to the ts-jest-keys-transformer.js file
+            astTransformers: ['<rootDir>/setupTests/ts-jest-keys-transformer.js'],
+        },
+        _: 'underscore',
+    },
+    testEnvironment: 'jest-environment-jsdom-fourteen',
+};
+
+// start
```
Comment (GermainBergeron) [inline packages/react-vapor/jest.config.js:27]: What are all those comments?

File: packages/react-vapor/src/components/listBox/ListBox.tsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ export class ListBox extends React.Component<IListBoxProps> {
                     key={item.value}
                     {...item}
                     onOptionClick={(option: IItemBoxProps) => this.onSelectItem(item, item.index)}
-                    selected={_.contains(this.props.selected, item.value)}
+                    active={_.contains(this.props.selected, item.value)} // here
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/listBox/ListBox.tsx:94]: I think you should revert this change

File: packages/react-vapor/package.json [hunk 3]
```diff
@@ -37,6 +35,10 @@
     },
     "dependencies": {
         "diff2html": "3.1.11",
+        "karma-rollup-plugin": "^0.2.4",
```
Comment (GermainBergeron) [inline packages/react-vapor/package.json:38]: are we still using karma?

File: packages/react-vapor/package.json [hunk 4]
```diff
@@ -63,6 +72,8 @@
         "underscore.string": "^3.3.5"
     },
     "devDependencies": {
+        "@babel/preset-react": "^7.10.4",
+        "@immowelt/jest-preset-react": "^3.1.0",
```
Comment (GermainBergeron) [inline packages/react-vapor/package.json:76]: babel? immowelt? are those used somewhere?

**Review by GermainBergeron — COMMENTED** [review]
You can probably do a cleanup at the same time (removing karma related files, removing packages that we don't use in the package.json, etc)

File: packages/react-vapor/setupTests/assetsTransformer.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const path = require('path');
+
+module.exports = {
+    process(src, filename, config, options) {
+        return 'module.exports = ' + JSON.stringify(path.basename(filename)) + ';';
```
Comment (rosalie-liu) [inline packages/react-vapor/setupTests/assetsTransformer.js:5]: Would it be easier to read `return `module.exports = ${JSON.stringify(path.basename(filename))};`;` ?

File: packages/react-vapor/jest.config.js (hunk 1 above)
Comment (FelixBlaisThon) [reply packages/react-vapor/jest.config.js:27]: Changes he made and how it fixed tests, but i dont know why it's in this pr and not the other
